delivery/models/db_models.py

```diff
@@ -138,3 +141,75 @@ def __repr__(self):
                 f"status: {self.delivery_status}, "
                 " }"
                 )
+
+class DDSDelivery(SQLAlchemyBase):
+
+    __tablename__ = "dds_deliveries"
+
+    dds_project_id = Column(String, primary_key=True)
+    ngi_project_name = Column(String)
+
+    date_started = Column(DateTime, nullable=False)
+    date_completed = Column(DateTime)
+    delivery_status = Column(Enum(DeliveryStatus), nullable=False)
+
+    def __repr__(self):
+        return (
+            "DDS Delivery: {"
+            f"dds_project: '{self.dds_project}', "
+            f"ngi_project_name: '{self.ngi_project_name}', "
+            f"date_started: {self.date_started}, "
+            f"date_completed: {self.date_completed}, "
+            f"delivery_status: {self.delivery_status}, "
+            " }"
+        )
+
+
+class DDSPut(SQLAlchemyBase):
+
+    __tablename__ = "dds_puts"
+
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    dds_project_id = Column(
+        String,
+        ForeignKey("dds_deliveries.dds_project_id"),
+        nullable=False)
+    dds_pid = Column(Integer, nullable=False)
+
+    delivery_source = Column(String, nullable=False)
+    delivery_path = Column(String, nullable=False)
+    destination = Column(String)
+
+    date_started = Column(DateTime, nullable=False)
+    date_completed = Column(DateTime)
+    delivery_status = Column(Enum(DeliveryStatus), nullable=False)
+
+    def __repr__(self):
+        return (
+            "DDS Put: {"
+            f"id: {self.id},"
+            f"dds_project: '{self.dds_project}', "
+            f"dds_pid: {self.dds_pid}, "
+            f"delivery_source: '{self.delivery_source}', "
+            f"delivery_path: '{self.delivery_path}', "
+            f"destination: '{self.destination}', "
+            f"date_started: {self.date_started}, "
+            f"date_completed: {self.date_completed}, "
+            f"delivery_status: {self.delivery_status}, "
+            " }"
+        )
+
+    def is_running(self):
```

Comment:
Ah, good point, this was written in March, I think at that time maybe I had a different idea for how this would work :thinking: Turns out I didn't need this in the end :shrug: I can remove it for now, if I find I need it again in the future I can put it back.